compiler/resolution/functionResolution.cpp

```diff
@@ -8732,10 +8732,34 @@ static Expr* resolveTypeOrParamExpr(Expr* expr) {
 *                                                                             *
 ************************************** | *************************************/
 
+// Non-normalizable expressions may nest at an arbitrary depth (because
+// they are not normalized), so we have to search upwards in the tree
+// to look for them.
+static Expr* handleNonNormalizableExpr(Expr* expr) {
+
+  // TODO: Ways to limit our search depth?
+  if (Expr* nonNormalRoot = partOfNonNormalizableExpr(expr)) {
+    if (CallExpr* call = toCallExpr(nonNormalRoot)) {
+      if (call->isPrimitive(PRIM_RESOLVES)) {
+        return preFold(call);
```

Comment:
Maybe add a comment saying that `preFold` will completely replace `PRIM_RESOLVES` primitive calls, and as a result other elements (like resolveExpr or postFold) are not needed here